From eee3a25c71f0598af63d2668ec0d4bc7845b6438 Mon Sep 17 00:00:00 2001
From: Vivek Kumar <vivek007.kumar@intel.com>
Date: Fri, 30 Jan 2015 17:40:00 -0800
Subject: [PATCH 274/275] atomisp2: Delay depth stream to avoid missing 1st
 sync

System timing changes introduced by Lollipop causes the ov680
stream to start before the ov8858 stream. In such cases the first
sync signal arrives in the middle of the first ov9728 frame
integration, causing the ov9728 to reset the frame and causing
the frame counts to go out of sync with ov8858.

Change-Id: I25e5675eff5a7518013951a9ab4a4b09cc6f3927
Tracked-On: https://jira01.devtools.intel.com/browse/IMINAN-19132
Signed-off-by: Vivek Kumar <vivek007.kumar@intel.com>
Signed-off-by: Soren Friis <soren.friis@intel.com>
Reviewed-on: https://icggerrit.ir.intel.com/40248
Reviewed-by: Vinod Govindapillai <vinod.govindapillai@intel.com>
Approver: Jukka O Kaartinen <jukka.o.kaartinen@intel.com>
Reviewed-by: Jukka O Kaartinen <jukka.o.kaartinen@intel.com>
Build: Wei Tao <wei.tao@intel.com>
Approver: Wei Tao <wei.tao@intel.com>
Reviewed-by: Wei Tao <wei.tao@intel.com>
Maintainer: Wei Tao <wei.tao@intel.com>
Integrator: Wei Tao <wei.tao@intel.com>
---
 .../external_drivers/camera/drivers/media/i2c/ov8858.c |  6 +++++-
 .../external_drivers/camera/drivers/media/i2c/ov8858.h |  2 +-
 .../media/pci/atomisp2/atomisp_driver/atomisp_ioctl.c  | 18 ++++++++----------
 3 files changed, 14 insertions(+), 12 deletions(-)

diff --git a/drivers/external_drivers/camera/drivers/media/i2c/ov8858.c b/drivers/external_drivers/camera/drivers/media/i2c/ov8858.c
index 2b69192..3feca1a 100644
--- a/drivers/external_drivers/camera/drivers/media/i2c/ov8858.c
+++ b/drivers/external_drivers/camera/drivers/media/i2c/ov8858.c
@@ -672,7 +672,11 @@ static int ov8858_g_comp_delay(struct v4l2_subdev *sd, unsigned int *usec)
 	if (ret || vts == 0)
 		vts = OV8858_DEPTH_VTS_CONST;
 
-	*usec = (exposure * 33333 / vts) - OV8858_DEPTH_COMP_CONST;
+	*usec = (exposure * 33333 / vts);
+	if (*usec >  OV8858_DEPTH_COMP_CONST)
+		*usec = *usec  - OV8858_DEPTH_COMP_CONST;
+	else
+		*usec = OV8858_DEPTH_COMP_CONST;
 
 	return 0;
 }
diff --git a/drivers/external_drivers/camera/drivers/media/i2c/ov8858.h b/drivers/external_drivers/camera/drivers/media/i2c/ov8858.h
index 847013a..b643d48 100644
--- a/drivers/external_drivers/camera/drivers/media/i2c/ov8858.h
+++ b/drivers/external_drivers/camera/drivers/media/i2c/ov8858.h
@@ -182,7 +182,7 @@ struct ov8858_vcm {
 
 #define MAX_FPS_OPTIONS_SUPPORTED		3
 
-#define OV8858_DEPTH_COMP_CONST			3200
+#define OV8858_DEPTH_COMP_CONST			2200
 #define OV8858_DEPTH_VTS_CONST			2573
 
 enum ov8858_tok_type {
diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_ioctl.c b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_ioctl.c
index 6d92f69..7b8a53a 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_ioctl.c
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_ioctl.c
@@ -1588,15 +1588,13 @@ int atomisp_stream_on_master_slave_sensor(struct atomisp_device *isp, bool isp_t
 			 isp->inputs[slave].camera->name);
 	}
 
-	if (isp_timeout) {
-		ret = v4l2_subdev_call(isp->inputs[master].camera, core,
-				       ioctl, ATOMISP_IOC_G_DEPTH_SYNC_COMP,
-				       &delay_slave);
-		if (ret)
-			dev_warn(isp->dev,
-				 "get depth sensor %s compensation delay failed.\n",
-				 isp->inputs[master].camera->name);
-	}
+	ret = v4l2_subdev_call(isp->inputs[master].camera, core,
+			       ioctl, ATOMISP_IOC_G_DEPTH_SYNC_COMP,
+			       &delay_slave);
+	if (ret)
+		dev_warn(isp->dev,
+			 "get depth sensor %s compensation delay failed.\n",
+			 isp->inputs[master].camera->name);
 
 	ret = v4l2_subdev_call(isp->inputs[master].camera,
 			       video, s_stream, 1);
@@ -1606,7 +1604,7 @@ int atomisp_stream_on_master_slave_sensor(struct atomisp_device *isp, bool isp_t
 		return -EINVAL;
 	}
 
-	if (isp_timeout && delay_slave != 0)
+	if (delay_slave != 0)
 		udelay(delay_slave);
 
 	ret = v4l2_subdev_call(isp->inputs[slave].camera,
-- 
1.9.1

